src/Toolchain.ml

```diff
@@ -0,0 +1,500 @@
+           match availablePackageManagers with
+           | Ok x -> x |> PackageManagerSet.of_list
+           | Error msg ->
+             Js.log2 "Error during availablePackageManagers()" msg;
+             PackageManagerSet.empty
+         in
+         let alreadyUsedPackageManagers =
+           match alreadyUsedPackageManagers with
+           | Ok x -> x
+           | Error msg ->
+             Js.log2 "Error during alreadyUsedPackageManagers()" msg;
+             PackageManagerSet.empty
+         in
+         Js.log "availablePackageManagers";
+         PackageManagerSet.iter
+           (fun x -> x |> PackageManager.toString |> Js.log)
+           availablePackageManagers;
+         Js.log "possiblyUsed";
+         PackageManagerSet.iter
+           (fun x -> x |> PackageManager.toString |> Js.log)
+           alreadyUsedPackageManagers;
+         match
+           PackageManagerSet.inter availablePackageManagers
+             alreadyUsedPackageManagers
+           |> PackageManagerSet.elements
+         with
+         | [] -> (
+           Js.log "Will lookup toolchain from global env";
+           match PackageManager.ofName projectRoot "<global>" with
+           | Ok t -> PackageManager.make ~env ~t
+           | Error msg -> Error msg |> Js.Promise.resolve )
+         | [ obviousChoice ] ->
+           Js.log2 "Toolchain detected" (PackageManager.toString obviousChoice);
+           PackageManager.make ~env ~t:obviousChoice
+         | multipleChoices -> (
+           let config = Vscode.Workspace.getConfiguration "ocaml" in
+           match
+             ( config
+               |. Vscode.WorkspaceConfiguration.get "packageManager"
+               |> Js.Nullable.toOption
+             , config
+               |. Vscode.WorkspaceConfiguration.get "toolChainRoot"
+               |> Js.Nullable.toOption )
+           with
+           | Some name, Some root ->
+             PackageManager.specOfName ~env ~name ~root:(Fpath.ofString root)
+           | Some name, None ->
+             PackageManager.specOfName ~env ~name ~root:projectRoot
+           | _ ->
+             (Window.showQuickPick
+                ( multipleChoices
+                |> List.map (fun pm -> PackageManager.toName pm)
+                |> Array.of_list )
+                (Window.QuickPickOptions.make ~canPickMany:false
+                   ~placeHolder:
+                     "Which package manager would you like to manage the \
+                      toolchain?"
+                   ()) [@bs])
+             |> Js.Promise.then_ (fun packageManager ->
+                    match Js.Nullable.toOption packageManager with
+                    | None ->
+                      Js.Promise.resolve
+                        (Error "showQuickPick() returned undefined")
+                    | Some packageManager ->
+                      Vscode.WorkspaceConfiguration.update config
+                        "packageManager" packageManager 2
```

Comment:
what's `2` here?

Reply:
An enumerated value as explained earlier. I'll try turning them to variants that can be compiled to ints